Add more pages bindings info (#7979)
cloudflare/cloudflare-docs: content/pages/framework-guides/deploy-a-nextjs-site.md

For the complete guide to deploying your first site to Cloudflare Pages, refer to the [Get started guide](/pages/get-started/).

## Use bindings in your Next.js application

> A [binding](/pages/platform/functions/bindings/) allows your application to interact with Cloudflare developer products, such as [KV](https://developers.cloudflare.com/workers/learning/how-kv-works/), [Durable Object](/workers/learning/using-durable-objects/), [R2](/r2/), and [D1](https://blog.cloudflare.com/introducing-d1/).

Reviewer: ```suggestion
Pages Functions allow you to take advantage of [Cloudflare Workers](/workers/) and add dynamic functionality to add to your Pages site. With Pages Functions, you can integrate with storage solutions, connect to third party services, and use server-side rendering with your favorite full stack frameworks. 

To use a Cloudflare storage resource, such as [KV](https://developers.cloudflare.com/workers/learning/how-kv-works/), [Durable Object](/workers/learning/using-durable-objects/), [R2](/r2/), and [D1](/d1/), with your Pages project:

1. Create a [Pages Function](/pages/platform/functions/get-started/).
2. Create a [binding](/pages/platform/functions/bindings/) that allows your Pages Function and storage resource to communicate.
```

> In Next.js you can add server-side code via [API Routes](https://nextjs.org/docs/api-routes/introduction) and [getServerSideProps](https://nextjs.org/docs/basic-features/data-fetching/get-server-side-props). In such server side code you can then use access bindings set for your application by accessing them via `process.env`.

Reviewer: ```suggestion
In Next.js, add server-side code via [API Routes](https://nextjs.org/docs/api-routes/introduction) and [getServerSideProps](https://nextjs.org/docs/basic-features/data-fetching/get-server-side-props). Then access bindings set for your application by accessing them in your code via `process.env`.
```

The following code block shows an example of accessing a KV namespace in Next.js.

```typescript
---
filename: src/index.tsx
highlight: [4, 5]
---
// ...

export async function getServerSideProps({ req }: GetServerSidePropsContext) => {

>   // the type KVNamespace comes from the @cloudflare/workers-types package

Reviewer: ```suggestion
  // the type `KVNamespace` comes from the @cloudflare/workers-types package
```
